### DISCUSSION TITLE
Keep <input> value if set on page load

### DISCUSSION THREAD

**Discussion by AndiLeni** [alpinejs/alpine#3899] [1. Help] [answered]
Hello,

it it possible to keep an input value for an <input> field if the input has a value on page load?
The usecase is the folowing: 
- the page gets rendered serverside
- initially the fields are empty, the user fills them which updates the x-model fields in a store
- when the user is ready, the form is submitted and gets shown to the user again with his inputs and the fields are disabled (user can check is input before final submission)

This causes trouble, because all x-models are iitialized and so the users inputs get overridden by alpine.
Can thsi somehow be prevented?

**Comment by SimoTod** [comment, marked as answer]
https://alpinejs.dev/directives/model#fill

**Reply by AndiLeni** [reply]
Thanks for pointing me to this.
It works fine for text inputs, but I can not see if this also works for checkboxes or radio buttons?

**Reply by SimoTod** [reply]
Did you try and it wasn't working?

**Reply by AndiLeni** [reply]
Yes, but it seems to be an issue on my end with the markup, because in a fiddle it works as expected.
